pkg/utils/kubeconf/kubelet_config_file.go

```diff
@@ -17,23 +17,40 @@ limitations under the License.
 package kubeconf
 
 import (
-	"os"
-
-	"github.com/ghodss/yaml"
+	"fmt"
 
 	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	kubeletconfigscheme "k8s.io/kubernetes/pkg/kubelet/apis/config/scheme"
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/configfiles"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
 )
 
 // GetKubeletConfigFromLocalFile returns KubeletConfiguration loaded from the node local config
+// based on https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L337
+// it fills empty fields with default values
```

Comment:
this looks indeed more correct, thanks for improving this function

Reply:
🎸